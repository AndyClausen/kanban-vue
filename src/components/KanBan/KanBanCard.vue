<template>
  <v-card class="mx-auto" max-width="344">
    <v-card-title>{{ card.title }}</v-card-title>
    <v-card-text>{{ card.description }}</v-card-text>
  </v-card>
</template>

<script setup lang="ts">
import type { Card } from "@/stores/app";

const { card } = defineProps<{ card: Card }>();
</script>
