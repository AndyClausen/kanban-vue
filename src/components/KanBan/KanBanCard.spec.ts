import { mount } from "@vue/test-utils";
import { describe, it, expect } from "vitest";
import KanBanCard from "./KanBanCard.vue";

const wrapper = mount(KanBanCard, {
  props: {
    card: {
      id: 1,
      title: "Card 1",
      description: "Description 1",
    },
  },
});

describe("KanBanCard", () => {
  it("renders the component", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
