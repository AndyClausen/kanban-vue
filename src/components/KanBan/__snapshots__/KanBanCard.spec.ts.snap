// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`KanBanCard > renders the component 1`] = `
"<v-card class="mx-auto" max-width="344">
  <v-card-title>Card 1</v-card-title>
  <v-card-text>Description 1</v-card-text>
</v-card>"
`;
